testing/tests/test_e2e/test_state.py

```diff
@@ -654,6 +658,143 @@ def _on_start(self, event: ops.StartEvent):
+        # (with an extra service and two checks added).
+        pytest.param(
+            {
+                'name': 'hello',
+                'title': 'Hello World',
+                'summary': 'A Hello World rock',
+                'description': 'This is an example of a Rockcraft project for a Hello World rock.',
+                'version': 'latest',
+                'base': 'bare',
+                'build-base': 'ubuntu@22.04',
+                'license': 'Apache-2.0',
+                'run-user': '_daemon_',
+                'environment': {
+                    'FOO': 'bar',
+                },
+                'services': {
+                    'hello': {
+                        'override': 'replace',
+                        'command': '/usr/bin/hello -t',
+                        'environment': {
+                            'VAR1': 'value',
+                            'VAR2': 'other value',
+                        },
+                    },
+                    'hello-echo': {
+                        'override': 'replace',
+                        'command': 'echo hello',
+                    },
+                },
+                'checks': {
+                    'chk1': {
+                        'override': 'merge',
+                        'exec': {
+                            'command': '/usr/bin/hello -c',
+                        },
+                    },
+                    'chk2': {
+                        'override': 'merge',
+                        'level': 'ready',
+                        'tcp': {
+                            'port': 8000,
+                        },
+                    },
+                },
+                'platforms': {
+                    'amd64': {},
+                    'armhf': {
+                        'build-on': ['armhf', 'arm64'],
+                    },
+                    'ibm': {
+                        'build-on': ['s390x'],
+                        'build-for': 's390x',
+                    },
+                },
+                'parts': {
+                    'hello': {
+                        'plugin': 'nil',
+                        'stage-packages': ['hello'],
+                    },
+                },
+            },
+            id='rockcraft-doc-example',
+        ),
+    ],
+)
+def test_layer_from_rockcraft(rockcraft: dict[str, Any]):
+    with tempfile.NamedTemporaryFile(mode='w', suffix='.yaml') as f:
+        yaml.safe_dump(rockcraft, f)
+        rockcraft_path = f.name
+        layer = layer_from_rockcraft(rockcraft_path)
+    assert layer.summary == rockcraft['summary']
+    if 'description' in rockcraft:
+        description = (
+            f'{rockcraft["description"]} (built from the rockcraft.yaml at {rockcraft_path})'
+        )
+    else:
+        description = f'(no description) (built from the rockcraft.yaml at {rockcraft_path})'
+    assert layer.description == description
```

Comment:
WDYT about loosening this up, maybe something like:
```suggestion
    if 'description' in rockcraft:
        assert description in layer.description
    assert rockcraft_path in description
```